src/components/checkbox/index.js

```diff
@@ -50,37 +69,46 @@ class Checkbox extends BaseComponent {
     /**
      * The selected icon color
      */
-    iconColor: PropTypes.string
+    iconColor: PropTypes.string,
   };
 
   generateStyles() {
     this.styles = createStyles(this.getThemeProps());
   }
 
+  check(value) {
+    const {isChecked} = this.state;
+
+    Animated.timing(isChecked, {
+      duration: 170,
+      easing: Easing.bezier(0.77, 0.0, 0.175, 1.0),
+      toValue: Number(value),
+      useNativeDriver: true,
+    }).start();
+  }
+
   onPress = () => {
     const {disabled} = this.getThemeProps();
     if (!disabled) {
       _.invoke(this.props, 'onValueChange', !this.props.value);
+      this.check(!this.props.value);
     }
   };
 
-  getContainerStyle() {
-    const {value, color, style: propsStyle, disabled} = this.getThemeProps();
-    const style = [this.styles.container];
-
-    if (value) {
-      if (disabled) {
-        style.push({backgroundColor: Colors.dark70, borderColor: Colors.dark70});
-      } else {
-        style.push(color ? {backgroundColor: color, borderColor: color} : this.styles.containerSelected);
-      }
-    } else if (disabled) {
-      style.push({borderColor: Colors.dark70});
-    } else if (color) {
-      style.push({borderColor: color});
+  getColors(element) {
```

Comment:
something with function looks over complicated.I believe we can do something simpler. 

the colors we use are either the passed color or the disabled color. and we use them both for the border and the background. 
create a simple function that returns the color (no need to pass element and create dynamic objects, it's making it less readable) 

now create a function for getting border style and for getting the background style. 
I would even call them getOuterStyle and getInnerStyle.